packages/azureblobstorage/src/Adaptor.js

```diff
@@ -0,0 +1,216 @@
+  return state;
+}
+
+// Destroy client
+function teardownClient(state) {
+  if (state.client) {
+    // Perform any other cleanup or disconnection logic if required
+    delete state.client;
+  }
+  return state;
+}
+
+// A basic Azure BlobServiceClient client is attached to state.client
+// but almost all operations require a ContainerClient. This function
+// just keeps things a little DRYer.
+function getContainerClient(state, containerName) {
+  const { client } = state;
+  return client.getContainerClient(containerName);
+}
+
+// DRY helper function, similar to getContainerClient, above,
+// but returns a BlockBlobClient
+function getBlockBlobClient(state, containerName, blobName) {
+  const containerClient = getContainerClient(state, containerName);
+  return containerClient.getBlockBlobClient(blobName);
+}
+
+/**
+ * Create a new container in Azure Blob Storage.
+ * @public
+ * @example
+ * createContainer('mycontainer')
+ * @function
+ * @param {string} containerName - Name of the container to create.
+ * @returns {Operation}
+ */
+export function createContainer(containerName) {
+  return async (state) => {
+    const containerClient = getContainerClient(state, containerName);
+    console.info(`Creating container '${containerName}'`);
+    const response = await containerClient.create();
+    console.debug(`Container '${containerName}' created successfully.`);
+    const nextState = composeNextState(state, response);
+    return nextState;
+  };
+}
+
+
+/**
+ * Check if a blob exists in Azure Blob Storage.
+ * @public
+ * @example
+ * checkBlobExists('mycontainer', 'myblob.txt')
+ * @function
+ * @param {string} containerName - Name of the container containing the blob.
+ * @param {string} blobName - Name of the blob to check.
+ * @returns {Operation}
+ */
+export function checkBlobExists(containerName, blobName) {
+  return async (state) => {
+    const blockBlobClient = getBlockBlobClient(state, containerName, blobName);
+    console.info(`Checking the existence of ${blobName} in container '${containerName}'`);
+    const blobExists = await blockBlobClient.exists();
+    let blobContainer = containerName; // renaming the property for consistency
+    console.debug(`Blob '${blobName}' in container '${blobContainer}' exists: ${blobExists}`);
+    const nextState = composeNextState(state, { blobName, blobContainer, blobExists });
+    return nextState;
+  };
+}
+
+/**
+ * Upload content to Azure Blob Storage.
+ * @public
+ * @example
+ * uploadBlob('mycontainer', 'myblob.txt', 'Hello, Azure Blob Storage!')
+ * @function
+ * @param {string} containerName - Name of the container where the blob will be uploaded.
+ * @param {string} blobName - Name of the blob to create or replace.
+ * @param {string} content - Content to upload.
+ * @param {object} uploadOptions - See BlockBlobUploadOptions in Azure Blob Storage docs
+ * @returns {Operation}
+ */
+export function uploadBlob(containerName, blobName, content, uploadOptions) {
```

Comment:
Since every operation takes a containerName as the first argument, is this something that should go on `state.configuration`? Would a single job ever need to use two containers? Is a credential/API key linked to or more containers?

The API for all functions would be simplified if `containerName` could go on state.config.

Reply:
updated to support this